code/modules/projectiles/guns/projectile.dm

```diff
@@ -124,10 +124,7 @@
 
 /obj/item/weapon/gun/projectile/process_chambered()
 	var/obj/item/ammo_casing/AC = getAC()
-	if(in_chamber)
-		return 1 //{R}
-	if(isnull(AC) || !istype(AC))
-		return
+	can_discharge()
```

Comment:
I moved the checks that tell the gun if it can fire or not from this line to the `can_discharge()` proc.
So I'm calling the proc in the place where the checks used to be.